packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -0,0 +1,107 @@
+
+/// A iOS style scrollbar.
+///
+/// A scrollbar indicates which portion of a [Scrollable] widget is actually
+/// visible.
+///
+/// To add a scrollbar to a [ScrollView], simply wrap the scroll view widget in
+/// a [CupertinoScrollbar] widget.
+///
+/// See also:
+///
+///  * [ListView], which display a linear, scrollable list of children.
+///  * [GridView], which display a 2 dimensional, scrollable array of children.
+///  * [Scrollbar], a Material Design scrollbar that dynamically adapts to the
+///    platform showing either an Android style or iOS style scrollbar.
+class CupertinoScrollbar extends StatefulWidget {
+  /// Creates an iOS style scrollbar that wraps the given [child].
+  ///
+  /// The [child] should be a source of [ScrollNotification] notifications,
+  /// typically a [Scrollable] widget.
+  const CupertinoScrollbar({
+    Key key,
+    @required this.child,
+  }) : super(key: key);
+
+  /// The subtree to place inside the [CupertinoScrollbar].
+  ///
+  /// This should include a source of [ScrollNotification] notifications,
+  /// typically a [Scrollable] widget.
+  final Widget child;
+
+  @override
+  _CupertinoScrollbarState createState() => new _CupertinoScrollbarState();
+
+  /// Returns a [ScrollbarPainter] visually styled like the iOS scrollbar.
+  static ScrollbarPainter buildCupertinoScrollbarPainter(TickerProvider vsync) {
+    return new ScrollbarPainter(
+      vsync: vsync,
+      thickness: _kScrollbarThickness,
+      mainAxisMargin: _kScrollbarMainAxisMargin,
+      crossAxisMargin: _kScrollbarCrossAxisMargin,
+      radius: _kScrollbarRadius,
+      minLength: _kScrollbarMinLength,
+      timeToFadeout: _kScrollbarTimeToFade,
+    )..color = _kScrollbarColor;
+  }
+}
+
+class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProviderStateMixin {
+  ScrollbarPainter _painter;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _painter ??= CupertinoScrollbar.buildCupertinoScrollbarPainter(this);
+    _painter.textDirection = Directionality.of(context);
+  }
+
+  bool _handleScrollNotification(ScrollNotification notification) {
+    if (notification is ScrollUpdateNotification ||
+        notification is OverscrollNotification) {
+      _painter.update(notification.metrics, notification.metrics.axisDirection);
+    } else if (notification is ScrollEndNotification) {
+      // On iOS, the scrollbar can only go away once the user lifted the finger.
+      _painter.scheduleFade();
+    }
+    return false;
+  }
+
+  @override
+  void dispose() {
+    _painter.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new NotificationListener<ScrollNotification>(
+      onNotification: _handleScrollNotification,
+      // TODO(ianh): Maybe we should try to collapse out these repaint
+      // boundaries when the scroll bars are invisible.
```

Comment:
you can remove this TODO (and the matching one on the other side), we're not likely to do this any time soon.

Reply:
This is easier to do now. Is it worth doing and reshaping the tree each time opacity comes in and out of = 0.0?

Reply:
I think we should just wait until we have a reason to do it.

Reply:
Done. PTAL